Two-body-scf-guess.py

```diff
@@ -0,0 +1,126 @@
+    i_list[0][0].append(np.zeros((m*2, m*2)))
+    i_list[-1][1].append(np.zeros((m*2, m*2)))
+    for s in i_list:
+        s[0].append(np.zeros((m*2, m*2)))
+        s[1].append(np.zeros((m*2, m*2)))
+    for i, dd_lists in enumerate(list_of_ddls):
+        for j, dens in enumerate(dd_lists):
+            if j < i:
+                i_list[i][0].append(dens)
+            if j >= i:
+                i_list[i][1].append(dens)
+    return i_list
+
+def add_interacts(i_list, m):
+    overlap = []
+    for s in i_list:
+        s0 = sum(s[0])[m:m*2, m:m*2]
+        s1 = sum(s[1])[0:m, 0:m]
+        ol = np.add(s0, s1)
+        overlap.append(ol)
+    return overlap
+
+def compute_diags(overlap, monomer_dens, a):
+    diags = []
+    for i, ol in enumerate(overlap):
+        diag = overlap[i] - ((a-2)*monomer_dens[i])
+        diags.append(diag)
+    return diags
+
+def rho_diagonal(diags, z_rho, m):
+    for i, dgs in enumerate(diags):
+        z_rho[i*m:(i+1)*m, i*m:(i+1)*m] = diags[i]
+    rho_d = z_rho
+    return rho_d
+
+def rho(list_of_ddls, rho_d, m):
+    for i, dd_lists in enumerate(list_of_ddls):
+        for j, dens in enumerate(dd_lists):
+            while j >= i:
+                rho_d[i*m:(i+1)*m, (j+1)*m:(j+2)*m] = list_of_ddls[i][j][0:m, m:m*2]
+                rho_d[(j+1)*m:(j+2)*m, i*m:(i+1)*m] = list_of_ddls[i][j][m:2*m, 0:m]
+                break
+    rho = rho_d
+    return rho
+rho = rho(list_of_ddls, rho_d, m)
+    
+def two_body_guess(mol, monomers, dimers, bs):
+    monomer_dens = make_monomer_dens(monomers, bs)
+    list_of_ddls = make_list_of_ddls(dimers, bs)
+    z_rho = make_z_rho(m, a)
+    ei_list = make_ei_list(list_of_ddls)
+    i_list = fill_ei_list(list_of_ddls, ei_list, m)
+    overlap = add_interacts(i_list, m)
+    diags = compute_diags(overlap, monomer_dens, a)
+    rho_d = rho_diagonal(diags, z_rho, m)
+    rho = rho(list_of_ddls, rho_d, m)
+    nitr = run_scf(mol, bs, rho)
+    return nitr
+
+if __name__=="__main__":
+    mol = [("O", 0.216486, -1.989005, 0.035297),
+       ("H", -0.603911, -1.505274, -0.033552),
+       ("H", 0.138672, -2.535875, 0.799566),
+       ("H", 1.505276, -0.603920, 0.033505),
+       ("O", 1.989004, 0.216483, -0.035296),
+       ("H", 2.535922, 0.138690, -0.799534),
+       ("O", -1.989001, -0.216487, -0.035297),
+       ("H", -2.535947, -0.138688, -0.799513),
+       ("H", -1.505273, 0.603920, 0.033493),
+       ("O", -0.216485, 1.989007, 0.035294),
+       ("H", -0.138694, 2.535879, 0.799565),
+       ("H", 0.603919, 1.505277, -0.033513)]
+
+    bs = "aug-cc-pvdz"
+    monomers = [mol[0:3], mol[3:6], mol[6:9], mol[9:12]]
+    ms = monomers
+    dimers = [[(ms[0]+ms[1]), (ms[0]+ms[2]), (ms[0]+ms[3])], [(ms[0]+ms[1]), (ms[1]+ms[2]), (ms[1]+ms[3])], 
+              [(ms[0]+ms[2]), (ms[1]+ms[2]), (ms[2]+ms[3])], [(ms[0]+ms[3]), (ms[1]+ms[3]), (ms[2]+ms[3])]]
+    
+    monomer_dens = make_monomer_dens(monomers, bs)
+    m,m = monomer_dens[0].shape
+    a = len(monomer_dens)
```

Comment:
Here are the two declaratives I was asking about that my functions depend on, and they are general meaning the user isn't supplying these.

Reply:
If you move them into the `two_body_guess` function then you can remove the remainder of lines 114 to 124 as it all appears in `two_body_guess`. Also I'd move the call to `run_scf` out of `two_body_guess` and return the guess instead of the number of iterations.